### ISSUE TITLE
not downloading all photos

### ISSUE THREAD

**Issue by kylesorkness** [robclouth/Facebook-Pix-Downloader#1]
Following the usage instructions, I viewed the “Photos of You” tab of my Facebook profile, scrolled all the way to the bottom so that all the thumbnails loaded, then clicked the FB Pix Downloader extension icon.

The extension downloaded only 305 of the 699 photos.

The 305 that downloaded were not the first/oldest 305 photos. In fact, I am not seeing any (obvious) pattern regarding which ones were successfully downloaded and which were skipped.

I would be happy to provide any other information that might help troubleshoot.

**Comment by robclouth** [comment]
Hey!
Hmm, I'm not exactly sure why that happened. Is it the same 305 photos each
time?

**Comment by kylesorkness** [comment]
No, it isn't.

I tried the extension three times using Chromium version 89.0.4389.114 on Debian 10.9. The first time, it downloaded those 305 photos. The second time, it downloaded only 289. (And, in case it matters: those 289 were a subset of the 305.) The third time, it downloaded only 35. (And those 35 were a subset of the 289.)

The third time, I checked Chromium's developer console and noticed the following error.

```
VM574:13 Uncaught ReferenceError: saveFbids is not defined
    at exportTaggedPhotos (<anonymous>:13:3)
    at <anonymous>:16:3
```

(The console shows other errors, but I don't know enough to know whether they're related to this extension, nor to know whether they contain personal information I shouldn't be posting publicly. But if those might be useful, let me know.)

I then tried the extension using Chrome version 89.0.4389.128 on macOS. It downloaded only 202 photos. (And, perhaps interestingly, those 202 were still a subset of the 305.)

I then asked a friend to try the extension using Chrome on Windows 10. It didn't download _any_ photos for her.

**Comment by kylesorkness** [comment]
A shot in the dark here (since I have no idea how this extension works): I noticed that another Chrome extension that aims to do the same thing, [Tagged Photo Exporter for Facebook](https://chrome.google.com/webstore/detail/tagged-photo-exporter-for/cicalpgkgmacgnbnnkhdgpnicipnihgo?hl=en-US), apparently recently stopped working for people. Just four days ago, in reply to a support question there on the Chrome Web Store, the developer said that "There was a recent Facebook API change that broke this extension" and that the extension was being updated accordingly.

Could that same API change be affecting FB Pix Downloader?

**Comment by robclouth** [comment]
Yeah that's probably it. I'll take a look today. Thanks for investigating!

**Comment by jubi23** [comment]
was this ever looked into? extension isn't currently working for me, and the facebook design did change recently i believe

**Comment by NievelKnievel** [comment]
Hey there, followed the instructions on your app page and nothing happens when I click the extension in the top right corner, any and all help appreciated, cheers!
